pkg/ccl/changefeedccl/changefeed.go

```diff
@@ -97,11 +97,32 @@ func makeSpansToProtect(
 	return spansToProtect
 }
 
-// initialScanFromOptions returns whether or not the options indicate the need
-// for an initial scan on the first run.
-func initialScanFromOptions(opts map[string]string) bool {
+// initialScanTypeFromOpts determines the type of initial scan the changefeed
+// should perform on the first run given the options provided from the user
+func initialScanTypeFromOpts(opts map[string]string) changefeedbase.InitialScanType {
 	_, cursor := opts[changefeedbase.OptCursor]
-	_, initialScan := opts[changefeedbase.OptInitialScan]
-	_, noInitialScan := opts[changefeedbase.OptNoInitialScan]
-	return (cursor && initialScan) || (!cursor && !noInitialScan)
+	initialScanType, initialScanSet := opts[changefeedbase.OptInitialScan]
+	_, initialScanOnlySet := opts[changefeedbase.OptInitialScanOnly]
+	_, noInitialScanSet := opts[changefeedbase.OptNoInitialScan]
```

Comment:
presumably, this relies on validation done prior to ensure that both initial+no_initial are not set?

Reply:
yeah it does, I will move the validation checks into this function.